src/Filter/CustomFieldsFilterService.php

```diff
@@ -17,12 +17,11 @@ public function __construct(CustomFieldRepoService $repo)
         $this->repo = $repo;
     }
 
-    public function applyFilter($filterform, QueryBuilder $qb, $firstRootAlias = null)
+    public function applyFilter($entityClass, $filterform, QueryBuilder $qb, $firstRootAlias = null)
     {
         if (!$firstRootAlias) {
                 $firstRootAlias = $qb->getRootAliases()[0];
         }
-
```

Comment:
$entityClass = $qb->getRootEntities()[0];

Reply:
great hint! Didn't know there is something like this...